types/ember/test/utils.ts

```diff
@@ -69,3 +69,19 @@ function testDefineProperty() {
         return `${this.firstName} ${this.lastName}`;
     }));
 }
+
+(function() { /* assign */
+    Ember.assign({}, { a: 'b'});
+    Ember.assign({}, { a: 'b'}).a; // $ExpectType string
+    Ember.assign({ a: 6 }, { a: 'b'}).a; // $ExpectType string
+    Ember.assign({ a: 6 }, {}).a; // $ExpectType number
+    Ember.assign({ b: 6 }, {}).a; // $ExpectError
```

Comment:
Would be good to add test cases for multiple sources.

Reply:
done